clarity/contracts/pool/yield-token-pool.clar

```diff
@@ -441,21 +441,49 @@
 )
 
 ;; Returns the fee of current x and y and make balance to 0.
-(define-public (collect-fees (the-aytoken <yield-token-trait>) (the-token <ft-trait>))
+(define-public (collect-fees (the-aytoken <ft-trait>) (the-token <ft-trait>))
     
     (let
         (
             (aytoken (contract-of the-aytoken))
+            (token (contract-of the-token))
             (pool (unwrap! (map-get? pools-data-map { aytoken: aytoken }) invalid-pool-err))
             (address (get fee-to-address pool))
             (fee-x (get fee-balance-aytoken pool))
             (fee-y (get fee-balance-token pool))
-            (fee-collector (get fee-to-address pool))
-            
+            (rebate-rate (unwrap-panic (contract-call? .alex-reserve-pool get-rebate-rate)))
+            (fee-x-rebate (unwrap! (contract-call? .math-fixed-point mul-down fee-x rebate-rate) math-call-err))
+            (fee-y-rebate (unwrap! (contract-call? .math-fixed-point mul-down fee-y rebate-rate) math-call-err))
+            (fee-x-net (unwrap! (contract-call? .math-fixed-point sub-fixed fee-x fee-x-rebate) math-call-err))
+            (fee-y-net (unwrap! (contract-call? .math-fixed-point sub-fixed fee-y fee-y-rebate) math-call-err))            
         )
-        (asserts! (is-eq contract-caller fee-collector) err-not-authorized)
-        (and (> fee-x u0) (unwrap! (contract-call? the-token transfer fee-x .alex-vault address none) transfer-x-failed-err))
-        (and (> fee-y u0) (unwrap! (contract-call? the-aytoken transfer fee-y .alex-vault address none) transfer-y-failed-err))
+
+        (and (> fee-x u0) 
+            (and 
+                ;; first transfer fee-x to this contract
+                (unwrap! (contract-call? the-aytoken transfer fee-x .alex-vault tx-sender none) transfer-x-failed-err)                
+                ;; convert rebate amount to alex and send to fee-to-address
```

Comment:
although temporary, isn't it better to change tx-sender to (as-contract tx-sender) ?